### DISCUSSION TITLE
Hydratation error

### DISCUSSION THREAD

**Discussion by rasti8** [vercel/next.js#48014] [Help] [answered]
### Summary

Hello, 
I ran into a hydration error saying that the DOM was different than what was rendered on the server. This is because I was using a conditional statement to render a custom div in my component that was depending on the window size. Basically, if I start with a mobile screen size, I didn't get any hydration error, but if I first run the code with a desktop screen size, I got the error.
So my question is when I use the use client tag shouldn't the component be rendered client-side so it doesn't give me this hydration error?

Cheers

PS: I solved this issue by using a dynamic import with the { ssr: false } option

**Comment by icyJoseph** [comment, marked as answer]
The `use client` directive doesn't mean a component only renders Server Side, it means it renders in both, at least on the Next.js case, that is to say that `use client` components are components we all have been using up to this point.

Think of rendering a page, but not getting the HTML for the parts that are marked as client code, that'd produce:

- Bad SEO
- Limit the browsing experience to JS enabled clients
- Potentially create large Layout Shift

In SSR-hybrid frameworks, where code runs both in Client and Server, one has to be careful with accessing Window, DOM, an other browser APIs.

Server Components are the special kind here, because they only render on the Server, and never leak their code to the client.

You might be confused with the 'client only' poisoning package, but this is so that a server component doesn't import a module which 
contains code that depends on window.

**Reply by rasti8** [reply]
Clear explanation! Thanks for the response!
